google/cloud/forseti/scanner/scanner_builder.py

```diff
@@ -51,60 +55,79 @@ def build(self):
-            if scanner.get('enabled'):
-                module_path = 'google.cloud.forseti.scanner.scanners.{}'
-
-                requirements_map = scanner_requirements_map.REQUIREMENTS_MAP
-                if not scanner.get('name') in requirements_map:
-                    log_message = (
-                        'Configured scanner is undefined '
-                        'in scanner requirements map : %s')
-                    LOGGER.error(log_message, scanner.get('name'))
-                    continue
-
-                module_name = module_path.format(
-                    scanner_requirements_map.REQUIREMENTS_MAP
-                    .get(scanner.get('name'))
-                    .get('module_name'))
-
-                try:
-                    module = importlib.import_module(module_name)
-                except (ImportError, TypeError, ValueError):
-                    LOGGER.exception('Unable to import %s\n', module_name)
-                    continue
-
-                class_name = (
-                    scanner_requirements_map.REQUIREMENTS_MAP
-                    .get(scanner.get('name'))
-                    .get('class_name'))
-                try:
-                    scanner_class = getattr(module, class_name)
-                except AttributeError:
-                    LOGGER.exception('Unable to instantiate %s', class_name)
-                    continue
-
-                # Simple way to find the path to folders directory no matter
-                # where forseti runs.
-                rules_path = self.scanner_configs.get('rules_path')
-                if rules_path is None:
-                    scanner_path = inspect.getfile(scanner_class)
-                    rules_path = scanner_path.split('/google/cloud/forseti')[0]
-                    rules_path += '/rules'
-
-                rules_filename = (scanner_requirements_map.REQUIREMENTS_MAP
-                                  .get(scanner.get('name'))
-                                  .get('rules_filename'))
-                rules = '{}/{}'.format(rules_path, rules_filename)
-                LOGGER.info('Initializing the rules engine:\nUsing rules: %s',
-                            rules)
-
-                scanner = scanner_class(self.global_configs,
-                                        self.scanner_configs,
-                                        self.service_config,
-                                        self.model_name,
-                                        self.snapshot_timestamp,
-                                        rules)
-                runnable_scanners.append(scanner)
+        if self.scanner_name == 'external_project_access_scanner':
+            scanner = self.make_scanner(external_project_access_scanner.
+                                        ExternalProjectAccessScanner,
+                                        'external_project_access_rules.yaml')
+            runnable_scanners.append(scanner)
+        elif not self.scanner_name:
```

Comment:
What if someone passing in a scanner name that doesn't exist?  Say, "foobar_scanner".  Would an empty `[ ]` runnable_scanner be okay?

Reply:
that is actually gonna be caught in the cli level, outputing: 
`(venv) angelsun@angelsun2:~/forseti-security$ forseti scanner run --scanner whateverScanner
Argument error: argument --scanner: invalid choice: 'whateverScanner' (choose from 'external_project_access_scanner').
usage: forseti scanner run [-h] [--scanner {external_project_access_scanner}]
`

Reply:
I removed the else statement, as it will not be needed since the cli handles the issue already. Please let me know if you have other recommendations